app/src/main/java/de/grobox/transportr/trips/search/DirectionsActivity.java

```diff
@@ -147,6 +152,26 @@ private void processIntent() {
 					to = viewModel.getWork().getValue();
 					if (to == null) new WorkPickerFragment().show(getSupportFragmentManager(), WorkPickerFragment.TAG);
 					break;
+				case INTENT_URI_FAVORITE:
+					from = null;
+					via = null;
+					to = null;
+					long uid = intent.getLongExtra("uid",0);
+
+					LiveData<List<FavoriteTripItem>> tripsLiveData;
+					tripsLiveData = viewModel.getFavoriteTrips();
+					tripsLiveData.observe(this, trips -> {
+						for (FavoriteTripItem trip : trips) {
+							if (trip!=null && trip.getUid()==uid)
+							{
+								searchFromTo(trip.getFrom(), trip.getVia(), trip.getTo());
+								tripsLiveData.removeObservers(this); // everything is done
+								break;
+							}
+						}
```

Comment:
What happens if the user deleted a favorite trip, but still has the shortcut on the launcher? How is this handled?

Reply:
The shortcut will just start a `DirectionsActivity` with empty source and destination fields. The observer won't find a matching trip in the list and do nothing.
Should we try to delete shortcuts when unmarking trips as favorite?

Reply:
I think when the observer doesn't find any trips, we could just show a Toast saying that this favorite was removed and that the user should remove the shortcut as well.